### PR TITLE
🧪 [testing improvement] Add comprehensive unit tests for Stack.Pop

### PR DESCRIPTION
**Pull request by gitrgoliveira**
🎯 **What:** Added comprehensive unit tests for the exported `Pop` method in the `Stack` struct.
📊 **Coverage:** The new tests cover LIFO (Last-In-First-Out) order, edge cases like popping from an empty stack, and complex mixed push/pop sequences.
✨ **Result:** Increased reliability of the `Stack` helper and improved test suite consistency by refactoring to use `testify/assert`.

---
*PR created automatically by Jules for task [896082066840130375](https://jules.google.com/task/896082066840130375) started by @gitrgoliveira*

### CODE REVIEW

File: internal/helper/tree_test.go [hunk 1]
```diff
@@ -448,18 +438,34 @@ func TestStack(t *testing.T) {
...
-			if stack.IsEmpty() {
-				t.Errorf("Stack should not be empty at iteration %d", i)
-			}
+			assert.False(t, stack.IsEmpty(), "Stack should not be empty at iteration %d", i)
 			popped := stack.Pop()
-			if popped.Val != int64(i) {
-				t.Errorf("Expected Val=%d, got Val=%d", i, popped.Val)
-			}
+			assert.NotNil(t, popped)
+			assert.Equal(t, int64(i), popped.Val)
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply internal/helper/tree_test.go:444]: @copilot apply changes based on [this feedback](https://github.com/gitrgoliveira/bracket-creator/pull/90#discussion_r3114027295)

File: internal/helper/tree_test.go [hunk 2]
```diff
@@ -412,32 +412,22 @@ func TestStack(t *testing.T) {
...
-		}
+		assert.NotNil(t, popped)
+		assert.Equal(t, int64(2), popped.Val, "Expected to pop node with Val=2")
 
 		popped = stack.Pop()
-		if popped.Val != 1 {
-			t.Errorf("Expected to pop node with Val=1, got Val=%d", popped.Val)
-		}
+		assert.NotNil(t, popped)
+		assert.Equal(t, int64(1), popped.Val, "Expected to pop node with Val=1")
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply internal/helper/tree_test.go:423]: @copilot apply changes based on [this feedback](https://github.com/gitrgoliveira/bracket-creator/pull/90#discussion_r3114027282)

File: internal/helper/tree_test.go [hunk 3]
```diff
@@ -412,32 +412,22 @@ func TestStack(t *testing.T) {
...
-			t.Error("Stack should not be empty after pushing")
-		}
+		assert.False(t, stack.IsEmpty(), "Stack should not be empty after pushing")
 
 		popped := stack.Pop()
-		if popped.Val != 2 {
-			t.Errorf("Expected to pop node with Val=2, got Val=%d", popped.Val)
-		}
+		assert.NotNil(t, popped)
+		assert.Equal(t, int64(2), popped.Val, "Expected to pop node with Val=2")
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply internal/helper/tree_test.go:420]: @copilot apply changes based on [this feedback](https://github.com/gitrgoliveira/bracket-creator/pull/90#discussion_r3114027274)

File: internal/helper/tree_test.go [hunk 4]
```diff
@@ -448,18 +438,34 @@ func TestStack(t *testing.T) {
...
+		assert.Equal(t, int64(2), popped.Val)
+
+		stack.Push(&Node{Val: 3})
+		popped = stack.Pop()
+		assert.NotNil(t, popped)
+		assert.Equal(t, int64(3), popped.Val)
+
+		popped = stack.Pop()
+		assert.NotNil(t, popped)
+		assert.Equal(t, int64(1), popped.Val)
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply internal/helper/tree_test.go:466]: @copilot apply changes based on [this feedback](https://github.com/gitrgoliveira/bracket-creator/pull/90#discussion_r3114027265)

File: internal/helper/tree_test.go [hunk 5]
```diff
@@ -448,18 +438,34 @@ func TestStack(t *testing.T) {
...
+		stack.Push(&Node{Val: 1})
+		stack.Push(&Node{Val: 2})
+		popped := stack.Pop()
+		assert.NotNil(t, popped)
+		assert.Equal(t, int64(2), popped.Val)
+
+		stack.Push(&Node{Val: 3})
+		popped = stack.Pop()
+		assert.NotNil(t, popped)
+		assert.Equal(t, int64(3), popped.Val)
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply internal/helper/tree_test.go:462]: @copilot apply changes based on [this feedback](https://github.com/gitrgoliveira/bracket-creator/pull/90#discussion_r3114027255)

File: internal/helper/tree_test.go [hunk 6]
```diff
@@ -448,18 +438,34 @@ func TestStack(t *testing.T) {
...
+	})
+
+	t.Run("mixed push and pop", func(t *testing.T) {
+		stack := Stack{}
+
+		stack.Push(&Node{Val: 1})
+		stack.Push(&Node{Val: 2})
+		popped := stack.Pop()
+		assert.NotNil(t, popped)
+		assert.Equal(t, int64(2), popped.Val)
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply internal/helper/tree_test.go:457]: @copilot apply changes based on [this feedback](https://github.com/gitrgoliveira/bracket-creator/pull/90#discussion_r3114027229)

**Comment by gitrgoliveira** [comment]
@copilot apply changes based on the comments in [this thread](https://github.com/gitrgoliveira/bracket-creator/pull/90#pullrequestreview-4143907389)
